### PR TITLE
WX-1409 Java 17

### PR DESCRIPTION
**Pull request by aednichols**

### CODE REVIEW

File: .github/workflows/chart_update_on_merge.yml [hunk 1]
```diff
@@ -23,9 +23,10 @@ jobs:
         repository: broadinstitute/cromwell
         token: ${{ secrets.BROADBOT_GITHUB_TOKEN }} # Has to be set at checkout AND later when pushing to work
         path: cromwell
-    - uses: olafurpg/setup-scala@v10
+    - uses: actions/setup-java@v4
       with:
-        java-version: adopt@1.11
```
Comment (aednichols) [inline .github/workflows/chart_update_on_merge.yml:28]: This action [is unmaintained](https://github.com/olafurpg/setup-scala/releases/tag/v14) and does not support our Java 17 distribution.

File: centaur/src/main/scala/centaur/api/DaemonizedDefaultThreadFactory.scala [hunk 2]
```diff
@@ -14,8 +14,7 @@ import java.util.concurrent.atomic.AtomicInteger
   * > main thread).
   */
 object DaemonizedDefaultThreadFactory extends ThreadFactory {
-  private val s = System.getSecurityManager
-  private val group = if (s != null) s.getThreadGroup else Thread.currentThread.getThreadGroup
```
Comment (aednichols) [inline centaur/src/main/scala/centaur/api/DaemonizedDefaultThreadFactory.scala:18]: `SecurityManager` is deprecated in 17. I honestly don't know what its role is here, but it certainly seems like the logic is just fine with the possibility of it being `null`.

File: cloud-nio/cloud-nio-spi/src/main/scala/cloud/nio/spi/UnixPath.scala [hunk 3]
```diff
@@ -67,7 +67,7 @@ final private[spi] case class UnixPath(path: String) extends CharSequence {
 
   def isAbsolute: Boolean = UnixPath.isAbsolute(path)
 
-  def isEmpty: Boolean = path.isEmpty
```
Comment (aednichols) [inline cloud-nio/cloud-nio-spi/src/main/scala/cloud/nio/spi/UnixPath.scala:70]: This is a name collision new in 17.

The initial compile error is that it needs an `override`. Adding the `override` results in a second error saying it overrides nothing!

File: docs/Configuring.md [hunk 4]
```diff
@@ -689,7 +689,7 @@ This limit may be configured via the configuration value:
 
 ```hocon
 yaml {
-  max-depth = 1000
```
Comment (aednichols) [inline docs/Configuring.md:692]: Depth of 1000 causes a stack overflow in 17.

I've never even heard of anyone using YAML inputs in real life, and I'm pretty confident 100 will be enough if I'm wrong.

File: project/Publishing.scala [hunk 5]
```diff
@@ -163,8 +163,7 @@ object Publishing {
   val additionalResolvers = List(
     broadArtifactoryResolver,
     broadArtifactoryResolverSnap,
-    Resolver.sonatypeRepo("releases")
```
Comment (aednichols) [inline project/Publishing.scala:166]: This was a deprecation

File: .github/set_up_cromwell_action/action.yml [hunk 6]
```diff
@@ -3,22 +3,23 @@
 name: 'Set Up Cromwell Steps'
 description: Specific steps that will set up git secrets, java, sbt, and Cromwell on the local machine. 
 inputs:
-  cromwell_repo_token: #As an input to this action, you are required to pass in a token that can be used to authenticate while checking out Cromwell.
+  cromwell_repo_token:
+    description: "As an input to this action, you are required to pass in a token that can be used to authenticate while checking out Cromwell."
```
Comment (aednichols) [inline .github/set_up_cromwell_action/action.yml:7]: IntelliJ believes that a `description` is required, so I put the comment into one.

File: runConfigurations/renderCiResources.run.xml [hunk 7]
```diff
@@ -7,8 +7,8 @@
     </option>
     <option name="tasks" value="renderCiResources" />
     <option name="useSbtShell" value="false" />
-    <option name="vmparams" value="-Xms512M -Xmx1024M -Xss1M -XX:+CMSClassUnloadingEnabled" />
```
Comment (aednichols) [inline runConfigurations/renderCiResources.run.xml:10]: This option was dropped in 17

File: scripts/docker-compose-mysql/README.md [hunk 8]
```diff
@@ -1 +0,0 @@
-Looking for the MySQL docs? Checkout the [Cromwell documentation](https://cromwell.readthedocs.io/en/develop/Configuring/#database)!
```
Comment (aednichols) [inline scripts/docker-compose-mysql/README.md:1]: The deleted files in `scripts/` required a Java update, but to my knowledge have not been used in years, so I decided to get rid of them.

File: supportedBackends/sfs/src/test/scala/cromwell/backend/sfs/SharedFileSystemJobExecutionActorSpec.scala [hunk 9]
```diff
@@ -1,430 +0,0 @@
...
-        "0"
-      } else {
-        import sys.process._
-        val proc = Seq("bash", "-c", s"sleep 2; echo 0 > ${jobPaths.returnCode}").run()
-        val pField = proc.getClass.getDeclaredField("p")
-        pField.setAccessible(true)
-        val p = pField.get(proc)
-        val pidField = p.getClass.getDeclaredField("pid")
-        pidField.setAccessible(true)
-        pidField.get(p).toString
```
Comment (aednichols) [inline supportedBackends/sfs/src/test/scala/cromwell/backend/sfs/SharedFileSystemJobExecutionActorSpec.scala:288]: This test relies on reflection in a way that does not work in 17.

We're already testing SFS a.k.a. Slurm in Centaur, which is preferred anyway.

**Review by jgainerdewar — APPROVED** [review]

**Review by JVThomas — APPROVED** [review]

**Pull request merged**